### PR TITLE
Kratos pm4ml providers

### PR DESCRIPTION
**Pull request by kalinkrustev**

### CODE REVIEW

File: terraform/gitops/k8s-cluster-config/app-deploy.tf [hunk 1]
```diff
@@ -200,6 +200,11 @@ locals {
   pm4ml_keycloak_realm_env_secret_map = { for key, pm4ml in local.pm4ml_var_map :
     "${var.pm4ml_oidc_client_secret_secret}-${key}" => var.pm4ml_oidc_client_secret_secret_key
   }
+
+  pm4ml_provider_secret_map = { for key, pm4ml in local.pm4ml_var_map :
```
Comment (dfry) [inline terraform/gitops/k8s-cluster-config/app-deploy.tf:204]: any reason why this is not just added to the same map as pm4ml_keycloak_realm_env_secret_map?

File: terraform/gitops/k8s-cluster-config/app-deploy.tf [hunk 2]
```diff
@@ -200,6 +200,11 @@ locals {
   pm4ml_keycloak_realm_env_secret_map = { for key, pm4ml in local.pm4ml_var_map :
     "${var.pm4ml_oidc_client_secret_secret}-${key}" => var.pm4ml_oidc_client_secret_secret_key
```
Comment (dfry) [inline terraform/gitops/k8s-cluster-config/app-deploy.tf:201]: does this go away with ory @kalinkrustev?

File: terraform/gitops/k8s-cluster-config/app-deploy.tf (hunk 1 above)
Comment (kalinkrustev) [reply terraform/gitops/k8s-cluster-config/app-deploy.tf:204]: The only reason is that I do not know the syntax to do this.

File: terraform/gitops/k8s-cluster-config/app-deploy.tf (hunk 2 above)
Comment (kalinkrustev) [reply terraform/gitops/k8s-cluster-config/app-deploy.tf:201]: This is in an area that I do not know yet. I see it relates to the experience API, where it is passed.

File: terraform/gitops/k8s-cluster-config/app-deploy.tf (hunk 1 above)
Comment (dfry) [reply terraform/gitops/k8s-cluster-config/app-deploy.tf:204]: ok, let's leave it as is and then clean up if necessary, especially if we end up removing the other client.

File: terraform/gitops/k8s-cluster-config/app-deploy.tf (hunk 2 above)
Comment (dfry) [reply terraform/gitops/k8s-cluster-config/app-deploy.tf:201]: we should review that with @geka-evk

File: terraform/gitops/generate-files/templates/ory/vault-secret.yaml.tpl [hunk 3]
```diff
@@ -162,6 +162,15 @@ spec:
             name: default
       name: kratosoidcsecret
       path: ${hubop_oidc_client_secret_secret_path}/${hubop_oidc_client_secret_secret}
+%{ for pm4ml in pm4mls ~}
```
[the remark this answers is not in the document]
Comment (kalinkrustev) [reply terraform/gitops/generate-files/templates/ory/vault-secret.yaml.tpl:165]: ok, will do

**Pull request merged**